core/state/journal_arbitrum.go

```diff
@@ -41,8 +41,11 @@ type EvictWasm struct {
 }
 
 func (ch EvictWasm) revert(s *StateDB) {
-	asm := s.GetActivatedAsm(ch.ModuleHash) // only happens in native mode
-	CacheWasmRust(asm, ch.ModuleHash, ch.Version, ch.Debug)
+	asm, err := s.TryGetActivatedAsm(ch.ModuleHash) // only happens in native mode
+	if err == nil && len(asm) == 0 {
+		//if we failed to get it - it's not in the current rust cache
+		CacheWasmRust(asm, ch.ModuleHash, ch.Version, ch.Debug)
```

Comment:
I'm not really sure what's happening here. If we failed to get asm, why are we caching that same asm? Don't we need to regenerate it or something?

Reply:
yeah, it's wrong. We shoul only cache it if we succeed.